Xiaohui/feat gicv3

buhenxihuan opened the pull request:
add gicv3 feature for arceos-hypervisor

buhenxihuan committed:
feat:support gicv3

buhenxihuan committed:
modify some code according to arm-gic-driver

buhenxihuan committed:
support for rk3588 smp

buhenxihuan committed:
bugfix:fix the gicr init bug accoring to the latest arm-gic-driver

aarkegz commented on configs/platforms/aarch64-qemu-virt.toml:
```diff
@@ -97,6 +98,8 @@ gicc-paddr = 0x0801_0000        # uint
 # GIC Distributor base address
 gicd-paddr = 0x0800_0000        # uint
 
+gicr-paddr = 0x080a_0000 
```
Add a comment here.

aarkegz commented on modules/axhal/Cargo.toml:
```diff
@@ -57,6 +58,7 @@ riscv_goldfish = { version = "0.1", optional = true }
 aarch64-cpu = "10.0"
 tock-registers = "0.9"
 arm_gicv2 = { git = "https://github.com/arceos-hypervisor/arm_gicv2" }
```
`arm_gicv2` is still a dependency even if `gicv3` feature is enabled, is this the desired behavior?

buhenxihuan replied:
adjusted

aarkegz commented on modules/axhal/src/platform/aarch64_rk3588j/dw_apb_uart.rs:
```diff
@@ -165,5 +165,5 @@ pub fn getchar() -> Option<u8> {
 /// UART simply initialize
 pub fn init_early() {
-    UART.lock().init();
+    // UART.lock().init();
```
Leave a comment for the reason of disabling UART init here please.

buhenxihuan replied:
On the rk3588j platform we use, uboot will first initialize the serial port, so we don't need to initialize it again. If other platforms do not have this feature, we need to uncomment this

aarkegz replied:
OK, thanks👌

aarkegz reviewed (changes requested):
Some small fixes are required.

buhenxihuan committed:
build(aarch64): Update platform configuration and interrupt controller dependencies

- Modify aarch64-qemu-virt platform configuration, add GIC Rdistributor base address
- Update axhal module dependencies:
- Remove arm_gicv2 direct dependency
- Add conditional dependency to use arm_gicv2 for non-GICv3 architectures
- Update other dependency versions
- In aarch64_rk3588j platform, keep UART initialization comments

aarkegz commented on modules/axhal/Cargo.toml:
```diff
@@ -20,7 +20,7 @@ tls = ["alloc"]
 uspace = ["paging"]
 default = []
-hv = ["paging", "cortex-a", "percpu/arm-el2", "page_table_entry/arm-el2", "arm_gicv2/el2", "dep:crate_interface"]
+hv = ["paging", "cortex-a", "percpu/arm-el2", "page_table_entry/arm-el2", "dep:crate_interface"]
```
`hv = [..., "arm_gicv2?/el2"]`, or "el2" will be included even if "hv" is not specified.

buhenxihuan replied:
adjusted

buhenxihuan committed:
build(AXHal): Update GICv2 dependency version and add configuration conditions

- Add GICv2 EL2 features for scenarios that support virtualization
- Add GICv2 base version for scenarios that do not support virtualization
- Optimize dependency versions through conditional compilation configuration

aarkegz reviewed (approved):
Mergable.
